driver-core/src/main/java/com/datastax/driver/core/HostConnectionPool.java

```diff
@@ -126,7 +124,7 @@ public void run() {
             }
             reusedConnection = null;
             connections.add(connection);
-            connectionFutures.add(handleErrors(setKeyspaceAsync(connectionFuture, connection, keyspace), initExecutor));
```

Comment:
I did like this the way it was because we know that the keyspace is properly set before using a core connection, but there was still always the reality that any newly spawned connection would need to have it's keyspace set in a blocking fashion which was a problem anyways.  [JAVA-893](https://datastax-oss.atlassian.net/browse/JAVA-893) will solve that though.